homeassistant/components/solarlog/sensor.py

```diff
@@ -71,20 +71,28 @@ async def async_setup_entry(hass, entry, async_add_entities):
 class SolarlogSensor(Entity):
     """Representation of a Sensor."""
 
-    def __init__(self, platform_name, sensor_key, data):
+    def __init__(self, entry_id, platform_name, sensor_key, data):
         """Initialize the sensor."""
         self.platform_name = platform_name
         self.sensor_key = sensor_key
         self.data = data
+        self.entry_id = entry_id
         self._state = None
 
         self._json_key = SENSOR_TYPES[self.sensor_key][0]
+        self._label = SENSOR_TYPES[self.sensor_key][1]
         self._unit_of_measurement = SENSOR_TYPES[self.sensor_key][2]
+        self._icon = SENSOR_TYPES[self.sensor_key][3]
+
+    @property
+    def unique_id(self):
+        """Return the unique id."""
+        return f"{self.entry_id}_{self._label}"
```

Comment:
```suggestion
        return f"{self.entry_id}_{self.sensor_key"
```

Reply:
why it doesnt matter much since unique_id can be any string. the sensor key is more of an ID format with underscores etc